src/main/java/com/scaleset/cfbuilder/beanstalk/ApplicationResourceLifecycleConfig.java

```diff
@@ -0,0 +1,11 @@
+package com.scaleset.cfbuilder.beanstalk;
+
+/**
+ Constructs a {@code ApplicationResourceLifecycleConfig} that specifies lifecycle settings for resources that belong
+ to an ElasticBeanstalk {@link Application} and the service role.
+
+ @see
+ <a href="https://docs.aws.amazon.com/IAM/latest/UserGuide/aws-properties-elasticbeanstalk-application-applicationresourcelifecycleconfig.html">Documentation Reference</a> */
+public class ApplicationResourceLifecycleConfig {
+    //TODO
```

Comment:
We should consider adding this in the future so we can consider the Beanstalk part of CloudFormation fully supported.